File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        seen = {}

        for i, num in enumerate(numbers):
            if not seen:
                seen[num] = i+1
                continue

            if (target - num) in seen.keys():
                return [seen[target-num], i+1]
            else:
                seen[num] = i+1

File: Data Structures & Algorithms/two-integer-sum/submission-5.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // dict for seen
        unordered_map<int, int> seen;

        // go through each num
        for(int i = 0; i < nums.size(); ++i){
            // if map not empty and i + j == target: return
            if(seen.find(target - nums[i]) != seen.end()){
                return {seen[target - nums[i]], i};
            }else{
                // else: add to the map and iterate
                seen[nums[i]] = i;
            }
        }
    }
};
